### PR TITLE
BWA-HaplotypeCaller Pipeline

### PR DESCRIPTION
**Pull request by jpfeil**
I added a new directory called gatk_pipeline. Now, everything starts from align-and-call which downloads shared reference files and then calls the other modules using a static dag. @heuermh  @fnothaft @almussel @jvivian @hannes-ucsc @benedictpaten

### CODE REVIEW

**Comment by jvivian** [comment]
Thanks @jpfeil !

@fnothaft, do you want to review this (or have me look at it) before merging?

File: src/toil_scripts/gatk_pipeline/align-and-call [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    """
+    Makes necessary reference indexes for BWA alignment
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+    sudo = input_args['sudo']
+    # Retrieve samples
+    return_input_paths(job, work_dir, shared_ids, 'ref.fa')
+    # Call: BWA
+    parameters = ["index", "ref.fa"]
+    docker_call(tool='quay.io/ucsc_cgl/bwa:0.7.12--dd5ac549b95eb3e5d166a5e310417ef13651994e',
```
Comment (heuermh) [inline src/toil_scripts/gatk_pipeline/align-and-call]: this step can take an awful long time, #73 instead requires the bwa indices as input parameters

File: src/toil_scripts/gatk_pipeline/align-and-call [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    job.addFollowOnJobFn(static_dag_declaration, shared_ids, input_args)
+
+
+def static_dag_declaration(job, shared_ids, input_args):
+    """
+    Now start alignment, preprocessing, and haplotype pipeline for each sample
+    """
+    # Define
+    align = job.wrapJobFn(bwa_germ.pipeline_start, shared_ids, input_args).encapsulate()
+    process = job.wrapJobFn(gatk_process.start_pipeline, shared_ids,
```
Comment (heuermh) [inline src/toil_scripts/gatk_pipeline/align-and-call]: Both ```pipeline_start``` and ```start_pipeline``` is confusing

**Comment by hannes-ucsc** [comment]
Setting the needs_work label. Feel free to remove it again if you think I'm in error and this is actually ready to go. I do remember asking @jpfeil to submit a PR with unfinished work just so it gets exposed to the team's collective expertise. Remove the label when you think it should be merged.

**Pull request closed without merging**